numpy/_core/src/npysort/x86_simd_qsort.dispatch.cpp

```diff
@@ -0,0 +1,124 @@
+/*@targets
+ * $maxopt $keep_baseline
+ * avx512_skx avx2
+ */
+// policy $keep_baseline is used to avoid skip building avx512_skx
+// when its part of baseline features (--cpu-baseline), since
+// 'baseline' option isn't specified within targets.
+
+#include "x86_simd_qsort.hpp"
+#ifndef __CYGWIN__
+
+#if defined(NPY_HAVE_AVX512_SKX)
+    #include "x86-simd-sort/src/avx512-32bit-qsort.hpp"
+    #include "x86-simd-sort/src/avx512-64bit-qsort.hpp"
+    #include "x86-simd-sort/src/avx512-64bit-argsort.hpp"
+#elif defined(NPY_HAVE_AVX2)
+    #include "x86-simd-sort/src/avx2-32bit-qsort.hpp"
+    #include "x86-simd-sort/src/avx2-64bit-qsort.hpp"
+#endif
+
+namespace np { namespace qsort_simd {
+
+#if defined(NPY_HAVE_AVX512_SKX) || defined(NPY_HAVE_AVX2)
+template<> void NPY_CPU_DISPATCH_CURFX(QSelect)(int32_t *arr, npy_intp num, npy_intp kth)
+{
+#if defined(NPY_HAVE_AVX512_SKX)
+    avx512_qselect(arr, kth, num);
+#else
+    avx2_qselect(arr, kth, num);
+#endif
```

Comment:
```suggestion
    x86_qselect(arr, kth, num);
```
Just create another template function, rather than keep duplicating the `#ifdef` directive.

Reply:
Good point. Fixed it.